### PR TITLE
Fire an event after content DOM created, before shown event for tooltip/popover 

### LINKED ISSUE #16124: Have tooltip/popover fire an event after content DOM created, before shown event
To assist users of data binding / templating libraries.
Refs #15674.

### PR DESCRIPTION
**Pull request by Johann-S**
Hi,

This PR will allow user to know when the tooltip template or popover is added to the DOM.
Fixes #16124.

/CC @cvrebert

### CODE REVIEW

**Comment by cvrebert** [comment]
Needs documentation.

**Comment by cvrebert** [comment]
A similar unit test needs to be added for Popovers too.

**Comment by Johann-S** [comment]
I added a unit test for the popover and documentation

**Comment by cvrebert** [comment]
@twbs-savage Retry

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 1c8612bc68dd3be78ee985182b304dad02bbcf85
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/55848318

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by Johann-S** [comment]
I don't know if it's good or not to do that but I haven't got any feedbacks for a while so...
PING :smile:

File: docs/_includes/js/popovers.html [hunk 1]
```diff
@@ -293,6 +293,10 @@ <h3 id="popovers-events">Events</h3>
           <td>hidden.bs.popover</td>
           <td>This event is fired when the popover has finished being hidden from the user (will wait for CSS transitions to complete).</td>
         </tr>
+        <tr>
+          <td>inserted.bs.popover</td>
+          <td>This event is fired after the <code>show</code> event when the popover template has been added to the DOM.</td>
```
Comment (cvrebert) [inline docs/_includes/js/popovers.html]: Should refer to the full event name (`show.bs.popover`) here

File: docs/_includes/js/tooltips.html [hunk 2]
```diff
@@ -260,6 +260,10 @@ <h3 id="tooltips-events">Events</h3>
           <td>hidden.bs.tooltip</td>
           <td>This event is fired when the tooltip has finished being hidden from the user (will wait for CSS transitions to complete).</td>
         </tr>
+        <tr>
+          <td>inserted.bs.tooltip</td>
+          <td>This event is fired after the <code>show</code> event when the tooltip template has been added to the DOM.</td>
```
Comment (cvrebert) [inline docs/_includes/js/tooltips.html]: Should refer to the full event name (`show.bs.tooltip`) here

File: js/tests/unit/popover.js [hunk 3]
```diff
@@ -259,4 +259,21 @@ $(function () {
...
+      .bootstrapPopover({
+        title: 'Test',
+        content: 'Test'
+      })
+      .on('inserted.bs.popover', function () {
+        assert.ok(true, 'inserted event fired')
+        done()
+      })
+
+    $popover.bootstrapPopover('show')
```
Comment (cvrebert) [inline js/tests/unit/popover.js]: This can just go after the `.on` call

**Comment by cvrebert** [comment]
The `inserted` event handlers in the unit tests should also assert that the tooltip/popover DOM exists.

**Comment by Johann-S** [comment]
Ok thanks for your feedbacks I'll work on it

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 174a54450d46d0c12deec1c3a6ae59088537588b
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/57624309

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 3d71eae3011b10daa558baf494b019925fee5f60
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/57627930

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by cvrebert** [comment]
Thanks for the revisions; LGTM.
@hnrch02 @XhmikosR LGTY?

**Comment by cvrebert** [comment]
@Johann-S Thanks!

**Comment by Johann-S** [comment]
You're welcome thanks for your feedbacks

**Pull request merged**